### ISSUE TITLE
LINODE: TTL does not support "Default" (0)

### ISSUE THREAD

**Issue by str4d** [StackExchange/dnscontrol#2440]
**Describe the bug**
I set up a `dnsconfig.js` matching the records for a domain I manage through Linode. All of the records in Linode have their TTL column set to "Default". I do not configure an explicit TTL in `dnsconfig.js` for any record, leaving it at the `dnscontrol` default.

When I run `dnscontrol preview`, every record is marked as a modification such as:
```
#1: MODIFY A example.com: (1.2.3.4 ttl=0) -> (1.2.3.4 ttl=300), Linode ID: NNN: {"type":"A","name":"","target":"1.2.3.4","ttl_sec":300}
```

**To Reproduce**
1. Configure a domain name to use Linode for DNS records.
2. Add a DNS A record to Linode with some label and IP, and leave the TTL as "Default".
3. Write a `dnsconfig.js` with the same A record label and IP, and no configured TTL.
4. `dnscontrol preview`

**Expected behavior**
No modifications should be proposed.

**Versions**
 - DNSControl version: 4.1.1 (Homebrew)
 - DNS Provider: `LINODE`

**Additional context**
The `LINODE` provider documentation has a [caveat](https://docs.dnscontrol.org/service-providers/providers/linode#caveats) that the TTLs must be from a specific list. [The file that it links to](https://github.com/linode/manager/blob/master/src/domains/components/SelectDNSSeconds.js) does not exist in the Linode source tree anymore. When I searched for an equivalent list, I found this one that explicitly includes the option `{ label: 'Default', value: 0 }`.

https://github.com/linode/manager/blob/07be31a25399c8583a55a3727b8a38e0a6e25cd7/packages/manager/src/features/Domains/DomainRecordDrawer.tsx#L320-L336

**Comment by str4d** [comment]
[Linode's own documentation](https://www.linode.com/docs/products/networking/dns-manager/guides/troubleshooting-dns/#set-the-time-to-live-or-ttl) confirms that the default TTL is still one day.

**Comment by TomOnTime** [comment]
CC @koesie10 

Interesting bug!

That raises an interesting question.  Should allowedTTLValues (https://github.com/StackExchange/dnscontrol/blob/d00804b4f51c5b95137b50b15c747493de1475d5/providers/linode/linodeProvider.go#L29) be changed to include `0` or should the fixTTL function turn 0 into 86400?  (I think the first option)

**Comment by str4d** [comment]
I personally agree that 0 should be allowed as a setting, because that is what the Linode API is returning when "Default" is set via their UI. If `fixTTL` turned 0 into 86400, that would still result in a `MODIFY` (it would be a functional no-op, but not a technical one).

The question though is whether `dnscontrol` has a good way to represent "Default" in its own UX. Do any other providers have a similar concept? How do they handle a TTL of 0?

**Comment by koesie10** [comment]
I agree that `0` should be added to the `allowedTTLValues`. I believe this should work and does not require any further modifications.

I have created [a PR ](https://github.com/StackExchange/dnscontrol/pull/2442) for this, which also updates the docs. I'm just finding conflicting information about what `0` actually means:
- [The DNS manager documentation](https://www.linode.com/docs/products/networking/dns-manager/guides/troubleshooting-dns/#set-the-time-to-live-or-ttl) says the default is 1 day
- [The API documentation](https://www.linode.com/docs/api/domains/#domains-list__responses) says "A value of 0 is equivalent to the default value of 1209600", which would be 14 days

I've taken the value from the API documentation since that's what the provider is using, but I'm not sure if that actually matches with what Linode does.

**Comment by str4d** [comment]
@koesie10 I think you were looking at the wrong part of that documentation.
<html><body>


data | array of objects
-- | --
expire_sec | integerThe amount of time in seconds that may pass before this Domain is no longer authoritative.Valid values are 0, 300, 3600, 7200, 14400, 28800, 57600, 86400, 172800, 345600, 604800, 1209600, and 2419200.Any other value is rounded up to the nearest valid value.A value of 0 is equivalent to the default value of 1209600.
-- | --
ttl_sec | integer“Time to Live” - the amount of time in seconds that this Domain’s records may be cached by resolvers or other domain servers.Valid values are 0, 300, 3600, 7200, 14400, 28800, 57600, 86400, 172800, 345600, 604800, 1209600, and 2419200.Any other value is rounded up to the nearest valid value.A value of 0 is equivalent to the default value of 86400.
-- | --



</body>
</html>

AFAICT the default is still 1 day for `ttl_sec`, not 14 days.

**Comment by str4d** [comment]
Fixed the comment as part of #2444.
